docusaurus/docs/how-to-guides/plugin-security.md

```diff
@@ -0,0 +1,114 @@
+---
+id: plugin-security
+title: Plugin security
+description: A guide for secure plugin development.
+keywords:
+  - grafana
+  - plugins
+  - security
+  - safety
+  - privacy
+  - plugin security
+---
+
+# Secure plugin development
+
+Building secure plugins for Grafana is essential to protecting user data, maintaining trust, and avoiding vulnerabilities that could compromise your Grafana plugin. This guide covers best practices for secure plugin development across the backend and frontend.
+
+## General best practices
+
+### Security audits and reviews
+
+Before submitting a plugin for publication, conduct security audits and reviews. Use tools like `gosec` for Go and `eslint` for TypeScript to catch common issues. Peer reviews should focus on identifying injection vulnerabilities, improper handling of sensitive data, and weak authentication.
+
+### Minimizing dependencies
+
+Keep dependencies to a minimum. Update packages regularly and remove unused ones. Use tools like `npm audit` and `go mod tidy` to identify outdated or vulnerable dependencies.
+
+### Secure coding practices
+
+- **Don’t hardcode secrets**: Avoid hardcoding API keys, tokens, or passwords in your code.
+- **Use least privilege**: Ensure your plugin uses the least privileged account or role when accessing resources.
+
+### Data handling
+
+- **Prevent data exposure**: Don’t expose sensitive data like user credentials, API tokens, or personally identifiable information (PII) through logs, requests, or the browser console.
```

Comment:
Probably should provide some example of how to do this. One way is to scan automatically for potentially exposed secrets with trufflehog. Which we also want to offer as part of the GH action checks: https://github.com/grafana/grafana-community-team/issues/224